### PR TITLE
feat: include additional policy properties in plain text output

### PR DESCRIPTION
**Pull request by thisislawatts**
## Pull Request Submission

Please check the boxes once done.

The pull request must:

- **Reviewer Documentation**
    - [x] follow [CONTRIBUTING](https://github.com/snyk/cli/blob/main/CONTRIBUTING.md) rules
    - [ ] be accompanied by a detailed description of the changes
    - [ ] contain a risk assessment of the change (Low | Medium | High) with regards to breaking existing functionality. A change e.g. of an underlying language plugin can completely break the functionality for that language, but appearing as only a version change in the dependencies.
    - [ ] highlight breaking API if applicable
    - [ ] contain a link to the automatic tests that cover the updated functionality.
    - [ ] contain testing instructions in case that the reviewer wants to manual verify as well, to add to the manual testing done by the author.
    - [ ] link to the link to the PR for the User-facing documentation
- **User facing Documentation**
    - [ ] update any relevant documentation in gitbook by submitting a gitbook PR, and including the PR link here
    - [ ] ensure that the message of the final single commit is descriptive and prefixed with either `feat:` or `fix:` , others might be used in rare occasions as well, if there is no need to document the changes in the release notes. The changes or fixes should be described in detail in the commit message for the changelog & release notes.
- **Testing**
    - [ ] Changes, removals and additions to functionality must be covered by acceptance / integration tests or smoke tests - either already existing ones, or new ones, created by the author of the PR.

## Pull Request Review

All pull requests must undergo a thorough review process before being merged. 
The review process of the code PR should include code review, testing, and any necessary feedback or revisions. 
Pull request reviews of functionality developed in other teams only review the given documentation and test reports. 

Manual testing will not be performed by the reviewing team, and is the responsibility of the author of the PR.

For Node projects: It’s important to make sure changes in `package.json` are also affecting `package-lock.json` correctly.

****************************If a dependency is not necessary, don’t add it.**************************** 

When adding a new package as a dependency, make sure that the change is absolutely necessary. We would like to refrain from adding new dependencies when possible.
Documentation PRs in gitbook are reviewed by Snyk's content team. They will also advise on the best phrasing and structuring if needed.

## Pull Request Approval

Once a pull request has been reviewed and all necessary revisions have been made, it is approved for merging into 
the main codebase. The merging of the code PR is performed by the code owners, the merging of the documentation PR 
by our content writers.

## What does this PR do?

Introduces a new set of formatting behaviours, documented in: `remediation-based-format-issues.spec.ts` based on new policy annotations. These will appear as:

```ts
// Vulnerability object:
{
  appliedPolicyRules: {
    annotation?: {
      value: string;
      reason?: string;
    },
    severityChange?: {
      newSeverity?: Severity;
      originalSeverity?: Severity;
      reason: string?
    };
    ignore?: {
      ignoredBy: user;
      created: Date;
      path: string[]
      source?: string;
      ignoredBy: IgnoredBy;
      created: string;
      expires?: string;
      reason: string;
      disregardIfFixable: boolean;
      reasonType: string;
    };
  }
}
```

During the development cycle we were running. `npx jest remediation-based-format-issues.spec.ts --watch`

## Example output:

```bash
    Issues to fix by upgrading dependencies:
    
      Upgrade django@1.6.1 to django@2.2.18 to fix
      ✗ Content Spoofing [Medium Severity][https://security.snyk.io/vuln/SNYK-PYTHON-DJANGO-72888] in django@1.6.1
        introduced by:
        django@1.6.1
    
    Issues with no direct upgrade or patch:
      ✗ Directory Traversal [Low Severity][https://security.snyk.io/vuln/SNYK-PYTHON-DJANGO-1066259] in django@1.6.1
        introduced by:
        django@1.6.1
        User note: Papercut
        Note reason: This vulnerability is a papercut and can be ignored
      This issue was fixed in versions: 2.2.18, 3.0.12, 3.1.6
```

**Note:** ANSI Control characters which set the colours are not included as part of this demo output here. Instead focus on the 2 new properites User note and Note Reason.

### CODE REVIEW

File: src/lib/formatters/types.ts [hunk 1]
```diff
@@ -12,6 +12,29 @@ export interface BasicVulnInfo {
...
+}
+
+export interface AppliedPolicyRules {
+  annotation?: {
+    value: string;
+    reason?: string;
+  };
+  severityChange?: {
+    newSeverity?: SEVERITY;
+    originalSeverity?: SEVERITY;
```
Comment (pmlyng) [inline src/lib/formatters/types.ts:27]: Perhaps still need to add the optional `reason?: string;` here?

Comment (thisislawatts) [reply src/lib/formatters/types.ts:27]: Good catch, added here: https://github.com/snyk/cli/pull/5142/commits/53fe35328ce9afac99514e5180a9a33cdc25a06a

File: test/jest/acceptance/snyk-test/human-formatted-output.spec.ts [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+      await project.readJSON('test-graph-results-with-annotation.json'),
+    );
+
+    const { code, stdout } = await runSnykCLI(`test`, {
+      cwd: project.path(),
+      env,
+    });
+
+    expect(code).toEqual(1);
+    expect(stripAnsi(stdout)).toContain('User note: This is a test user note');
```
Comment (j-luong) [inline test/jest/acceptance/snyk-test/human-formatted-output.spec.ts:63]: question: should we test for the other new fields?
```
    (note ? `${chalk.bold('\n    Note')}:\n    ${note}` : '') +
    (severityReason ? '\n    Severity reason: ' + severityReason : '') +
    (userNote ? '\n    User note: ' + userNote : '') +
    (userNoteReason ? '\n    Note reason: ' + userNoteReason : '')
```

**Review by j-luong — CHANGES_REQUESTED** [review]
Just need to test the other new fields in the acceptance test

File: test/jest/acceptance/snyk-test/human-formatted-output.spec.ts (hunk 2 above)
Comment (thisislawatts) [reply test/jest/acceptance/snyk-test/human-formatted-output.spec.ts:63]: We do have those covered in the unit tests `test/jest/unit/lib/formatters/remediation-based-format-issues.spec.ts` added in this changeset. So the acceptance test we have added verifies the full stack but not all of the edges cases. This seems reasonable to balance speed and correctness.

**Review by j-luong — APPROVED** [review]

**Pull request merged**